### PR TITLE
add wt/4.3.1

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **wt/4.3.1**

this is a port of https://github.com/bincrafters/conan-wt

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/wt/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            self._cmake.definitions['OPENSSL_LIBRARIES'] = ';'.join(self.deps_cpp_info['openssl'].libs + self.deps_cpp_info['openssl'].system_libs)
+            self._cmake.definitions['OPENSSL_INCLUDE_DIR'] = ';'.join(self.deps_cpp_info['openssl'].include_paths)
+            self._cmake.definitions['OPENSSL_FOUND'] = True
+        if self.settings.os == 'Windows':
+            self._cmake.definitions['CONNECTOR_FCGI'] = False
+            self._cmake.definitions['CONNECTOR_ISAPI'] = self.options.connector_isapi
+        else:
+            self._cmake.definitions['CONNECTOR_FCGI'] = self.options.connector_fcgi
+            self._cmake.definitions['CONNECTOR_ISAPI'] = False
+            self._cmake.definitions['CMAKE_POSITION_INDEPENDENT_CODE'] = self.options.fPIC
```
Comment (uilianries) [inline recipes/wt/all/conanfile.py:106]:
```suggestion
```

File: recipes/wt/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
+cmake_minimum_required(VERSION 2.8.11)
+
+project(test_package)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
```
Comment (uilianries) [inline recipes/wt/all/test_package/CMakeLists.txt:6]:
```suggestion
```

File: recipes/wt/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+file(GLOB SOURCE_FILES *.cpp)
+
+add_executable(${PROJECT_NAME} ${SOURCE_FILES})
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/wt/all/test_package/CMakeLists.txt:14]:
```suggestion
target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
```

File: recipes/wt/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        tools.replace_in_file(os.path.join(self._source_subfolder, 'CMakeLists.txt'), 'find_package(OpenSSL)', '#find_package(OpenSSL)')
```
Comment (uilianries) [inline recipes/wt/all/conanfile.py:111]:
```suggestion
```
it could use cmake_find_package generator I think

**Comment by uilianries** [comment]
I tried to use Wt some years ago, but I have to say that was not productive.

File: recipes/wt/all/conanfile.py (hunk 4 above)
Comment (ericLemanissier) [reply recipes/wt/all/conanfile.py:110]: I tried it in several ways, but the variables set by conan generated cmake files do not match what wt expects

File: recipes/wt/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        self._cmake.definitions['ENABLE_SSL'] = self.options.with_ssl
+        self._cmake.definitions['ENABLE_HARU'] = self.options.with_haru
+        self._cmake.definitions['ENABLE_PANGO'] = self.options.with_pango
+        self._cmake.definitions['ENABLE_SQLITE'] = self.options.with_sqlite
+        self._cmake.definitions['ENABLE_POSTGRES'] = self.options.with_postgres
+        self._cmake.definitions['ENABLE_FIREBIRD'] = self.options.with_firebird
+        self._cmake.definitions['ENABLE_MYSQL'] = self.options.with_mysql
+        self._cmake.definitions['ENABLE_MSSQLSERVER'] = self.options.with_mssql
+        self._cmake.definitions['ENABLE_QT4'] = self.options.with_qt4
+        self._cmake.definitions['ENABLE_LIBWTTEST'] = self.options.with_test
```
Comment (madebr) [inline recipes/wt/all/conanfile.py:84]: This seems to be a selftest:
https://github.com/emweb/wt/blob/34addc77d805916a9aa41c9a2beac15c4a49899d/test/CMakeLists.txt#L1

File: recipes/wt/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+        "with_sqlite": [True, False],
+        "with_postgres": [True, False],
+        "with_firebird": [True, False],
+        "with_mysql": [True, False],
+        "with_mssql": [True, False],
+        "with_qt4": [True, False],
+        "with_test": [True, False],
+        "with_dbo": [True, False],
+        "with_opengl": [True, False],
+        "with_unwind": [True, False],
```
Comment (madebr) [inline recipes/wt/all/conanfile.py:32]: Some of these are missing in `requirements`

File: recipes/wt/all/conanfile.py (hunk 5 above)
Comment (ericLemanissier) [reply recipes/wt/all/conanfile.py:84]: it is used in selftest, but it's a set of class part of public api, which can be reused independantly of self tests: https://github.com/emweb/wt/blob/247fdd1f1e8ea407fd8a65fa8de7eee21f0d7daa/CMakeLists.txt#L43 
https://webtoolkit.eu/wt/doc/reference/html/classWt_1_1Test_1_1WTestEnvironment.html

Comment (madebr) [reply recipes/wt/all/conanfile.py:84]: Indeed. Thanks!
Users should be able to test their servers.

File: recipes/wt/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions['CONFIGDIR'] = os.path.join(self.package_folder, 'bin')
+        self._cmake.definitions['SHARED_LIBS'] = self.options.shared
+        self._cmake.definitions['BUILD_EXAMPLES'] = False
+        self._cmake.definitions['BUILD_TESTS'] = False
+        self._cmake.definitions['ENABLE_SSL'] = self.options.with_ssl
+        self._cmake.definitions['ENABLE_HARU'] = self.options.with_haru
```
Comment (uilianries) [inline recipes/wt/all/conanfile.py:106]: haru, pango, firebird and qt, we don't packages for them. I don't know if we should enable them from the system, as you know, it's hard to track any system version. I would prefer them until having available;

Comment (ericLemanissier) [reply recipes/wt/all/conanfile.py:106]: So you want me to remove these 4 options, and always disable the corresponding part, Do I understand correctly ?

Comment (uilianries) [reply recipes/wt/all/conanfile.py:106]: That's one option. Another one is raising ConanInvalidConfiguration when enabled.

File: recipes/wt/all/conanfile.py (hunk 4 above)
Comment (SSE4) [reply recipes/wt/all/conanfile.py:152]: that's mostly expected, as we're not yet providing fully compatible **find_package(OpenSSL)**, mostly due to the components - we have to expose something like **OpenSSL::Crypto** and **OpenSSL::SSL**, this is something to re-visit later.

**Pull request merged**